src/module-config/module-config.ts

```diff
@@ -7,35 +7,43 @@ const configs: object[] = [];
 const schemas = {};
 
 export function defineConfigSchema(moduleName, schema) {
-  // console.log( "defineConfigSchema received schema for " + moduleName + ": " + JSON.stringify(schema));
   schemas[moduleName] = schema;
 }
 
 export function provide(config) {
-  // console.log("provide recieved config " + JSON.stringify(config));
   configs.push(config);
 }
 
-let importMapConfigHasBeenAdded = false;
-export async function getConfig(moduleName: string): Promise<any> {
-  // Get config file from import map and prepend it to `configs`
-  if (!importMapConfigHasBeenAdded) {
-    let importMapConfigExists;
-    try {
-      System.resolve("config-file");
-      importMapConfigExists = true;
-    } catch {
-      importMapConfigExists = false;
-    }
+// We cache the Promise that loads the import mapped config file
+// so that we can be sure to only call it once
+let getImportMapConfigPromise;
+export async function getConfig(moduleName: string): Promise<ConfigObject> {
+  if (!getImportMapConfigPromise) {
+    getImportMapConfigPromise = getImportMapConfigFile();
+  }
+  return getImportMapConfigPromise.then(() => getConfigForModule(moduleName));
```

Comment:
Either use `await`, or don't make it `async`! 😄 

Async/await is syntactic sugar on top of promises. The function we have here is the worst of both worlds. Here's the async/await way of doing this:

```js
export async function getConfig(moduleName: string): Promise<ConfigObject> {
  if (!getImportMapConfigPromise) {
    getImportMapConfigPromise = getImportMapConfigFile();
  }
  await getImportMapConfigPromise;
  return await getConfigForModule(moduleName);
}
```